### PR TITLE
Apple: Shadowsocks tunnel

### PR DESCRIPTION
**Pull request by alalamav**
* Updates the Apple frameworks to use the Shadowsocks tunnel introduced in #26.
* Fixes Android documentation.

### CODE REVIEW

File: outline/apple/tun2socks.go [hunk 1]
```diff
@@ -47,18 +47,22 @@ func init() {
...
 // `isUDPEnabled` indicates whether the tunnel and/or network enable UDP proxying.
 //
 // Sets an error if the tunnel fails to connect.
-func ConnectSocksTunnel(tunWriter TunWriter, host string, port int, isUDPEnabled bool) (OutlineTunnel, error) {
-	if tunWriter == nil || host == "" || port <= 0 || port > 65535 {
-		return nil, errors.New("Must provide a TunWriter, a valid SOCKS proxy host and port")
+func ConnectShadowsocksTunnel(tunWriter TunWriter, host string, port int, password, cipher string, isUDPEnabled bool) (OutlineTunnel, error) {
+	if tunWriter == nil {
+		return nil, errors.New("Must provide a TunWriter")
+	} else if port <= 0 || port > 65535 {
```
Comment (bemasc) [inline outline/apple/tun2socks.go]: nit: consider using math.MaxUint16

File: outline/apple/tun2socks.go [hunk 2]
```diff
@@ -47,18 +47,22 @@ func init() {
...
+		return nil, errors.New("Must provide a valid port number")
```
Comment (bemasc) [inline outline/apple/tun2socks.go]: nit: consider using `fmt.Errorf` to include the invalid port number.

File: outline/apple/tun2socks.go (hunk 1 above)
Comment (alalamav) [reply outline/apple/tun2socks.go]: Done.

File: outline/apple/tun2socks.go (hunk 2 above)
Comment (alalamav) [reply outline/apple/tun2socks.go]: Done.

**Pull request merged**